lib/core/include/stream_factory_utility.hpp

```diff
@@ -128,7 +128,61 @@ namespace irods::experimental::io
 
             return d;
         };
-    }
+    } // make_dstream_factory
+
+    /// Creates a factory that produces managed_dstream objects.
+    ///
+    /// \param[in] _conn_pool    \parblock
+    ///                          The connection pool serving connections for each stream object created
+    ///                          by the factory. The connection pool is not owned by the generated factory
+    ///                          and must out live any streams produced by the factory.
+    ///                          \endparblock
+    /// \param[in] _logical_path The full path to a data object. All streams will point to this data object.
+    /// \param[in] _replica_id   The replica number or resource name used to identify a specific replica.
+    ///
+    /// \tparam ReplicaIdentifier \parblock
+    /// The type of the information used to identify a specific replica. The following
+    /// types are supported:
+    ///     - irods::experimental::io::replica_number
+    ///     - irods::experimental::io::leaf_resource_name
+    ///     - irods::experimental::io::root_resource_name
+    /// \endparblock
+    ///
+    /// \return A new factory function.
+    ///
+    /// \since 4.2.9
+    template <typename ReplicaIdentifier>
+    auto make_dstream_factory(irods::connection_pool& _conn_pool,
+                              std::string _logical_path,
+                              ReplicaIdentifier&& _replica_id)
+    {
+        return [&_conn_pool, p = std::move(_logical_path), repl_id = std::forward<ReplicaIdentifier>(_replica_id)]
```

Comment:
should `std::forward` be used here or does a normal capture by reference maintain the type information for forwarding later.

Reply:
through experimentation against clang and gcc, capturing by reference is the way to go.